community/kernel/src/main/java/org/neo4j/kernel/internal/GraphDatabaseAPI.java

```diff
@@ -48,6 +52,43 @@
      */
     URL validateURLAccess( URL url ) throws URLAccessValidationError;
 
-    @Deprecated
     String getStoreDir();
+
+    /**
+     * Returns all relationship types currently in the underlying store.
+     * Relationship types are added to the underlying store the first time they
+     * are used in a successfully commited {@link Node#createRelationshipTo
+     * node.createRelationshipTo(...)}. Note that this method is guaranteed to
+     * return all known relationship types, but it does not guarantee that it
+     * won't return <i>more</i> than that (e.g. it can return "historic"
+     * relationship types that no longer have any relationships in the node
+     * space).
+     *
+     * @return all relationship types in the underlying store
+     */
+    ResourceIterable<RelationshipType> getAllRelationshipTypesInUseAndNot();
```

Comment:
i would give names like `getAllRelationshipTypes` to methods that actually return all rel. types (like in our case) and rename method in GraphDatabaseService that returns only used one.